array/array.mbt

```diff
@@ -16,9 +16,8 @@ pub fn iter[T](self : Array[T]) -> Iter[T] {
   Iter::new(
     fn(yield) {
       for v in self {
-        if yield(v) == IterEnd {
-          break IterEnd
-        }
+        guard let IterContinue = yield(v) else { x => break x }
```

Comment:
cc @Yoorkin  @lynzrand we need check if the coverage works with the new guard let